drivers/video/video_esp32_dvp.c

```diff
@@ -0,0 +1,425 @@
+
+	return video_get_ctrl(cfg->sensor_dev, cid, value);
+}
+
+static void video_esp32_cam_ctrl_init(const struct device *dev)
+{
+	const struct video_esp32_config *cfg = dev->config;
+	struct video_esp32_data *data = dev->data;
+
+	LCD_CAM.cam_ctrl.cam_stop_en = 0;
+	LCD_CAM.cam_ctrl.cam_vsync_filter_thres = 4;
+	LCD_CAM.cam_ctrl.cam_byte_order = cfg->invert_byte_order;
+	LCD_CAM.cam_ctrl.cam_bit_order = cfg->invert_bit_order;
+	LCD_CAM.cam_ctrl.cam_line_int_en = 0;
+	LCD_CAM.cam_ctrl.cam_vs_eof_en = 1;
+
+	LCD_CAM.cam_ctrl1.val = 0;
+	LCD_CAM.cam_ctrl1.cam_rec_data_bytelen = 0;
+	LCD_CAM.cam_ctrl1.cam_line_int_num = 0;
+	LCD_CAM.cam_ctrl1.cam_clk_inv = cfg->invert_pclk;
+	LCD_CAM.cam_ctrl1.cam_vsync_filter_en = 1;
+	LCD_CAM.cam_ctrl1.cam_2byte_en = cfg->enable_16_bit;
+	LCD_CAM.cam_ctrl1.cam_de_inv = cfg->invert_de;
+	LCD_CAM.cam_ctrl1.cam_hsync_inv = cfg->invert_hsync;
+	LCD_CAM.cam_ctrl1.cam_vsync_inv = cfg->invert_vsync;
+	LCD_CAM.cam_ctrl1.cam_vh_de_mode_en = 0;
+
+	LCD_CAM.cam_rgb_yuv.val = 0;
+}
+
+static int esp32_init(const struct device *dev)
+{
+	const struct video_esp32_config *cfg = dev->config;
+	struct video_esp32_data *data = dev->data;
+	int ret = 0;
+
+	data->config = cfg;
+
+	k_fifo_init(&data->fifo_in);
+	k_fifo_init(&data->fifo_out);
+
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("video pinctrl setup failed (%d)", ret);
+		return ret;
+	}
+
+	video_esp32_cam_ctrl_init(dev);
+
+	if (!device_is_ready(cfg->dma_dev)) {
+		return -ENODEV;
+	}
+
+	LOG_DBG("esp32 video driver loaded");
+
+	return ret;
+}
+
+static const struct video_driver_api esp32_driver_api = {
+	/* mandatory callbacks */
+	.set_format = video_esp32_set_fmt,
+	.get_format = video_esp32_get_fmt,
+	.stream_start = video_esp32_stream_start,
+	.stream_stop = video_esp32_stream_stop,
+	.get_caps = video_esp32_get_caps,
+	/* optional callbacks */
+	.enqueue = video_esp32_enqueue,
+	.dequeue = video_esp32_dequeue,
+	.flush = NULL,
+	.set_ctrl = video_esp32_set_ctrl,
+	.get_ctrl = video_esp32_get_ctrl,
+	.set_signal = NULL,
+};
+
+PINCTRL_DT_INST_DEFINE(0);
+
+static const struct video_esp32_config esp32_config = {
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+	.sensor_dev = DEVICE_DT_GET(DT_INST_PHANDLE(0, sensor)),
+	.dma_dev = ESP32_DT_INST_DMA_CTLR(0, rx),
+	.rx_dma_channel = DT_INST_DMAS_CELL_BY_NAME(0, rx, channel),
+	.enable_16_bit = DT_INST_PROP(0, enable_16bit_mode),
+	.invert_bit_order = DT_INST_PROP(0, invert_bit_order),
+	.invert_byte_order = DT_INST_PROP(0, invert_byte_order),
+	.invert_pclk = DT_INST_PROP(0, invert_pclk),
+	.invert_de = DT_INST_PROP(0, invert_de),
+	.invert_hsync = DT_INST_PROP(0, invert_hsync),
+	.invert_vsync = DT_INST_PROP(0, invert_vsync)};
+
+static struct video_esp32_data esp32_data = {0};
+
+DEVICE_DT_INST_DEFINE(0, &esp32_init, PM_DEVICE_DT_INST_GET(idx), &esp32_data, &esp32_config,
+		      POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY, &esp32_driver_api);
```

Comment:
Out of curiosity, are there any ESP32 with this driver and several instance of this peripheral?
If so, this could eventually be put under a `DT_INST_FOREACH_STATUS_OKAY(VIDEO_ESP32_DEVICE_DEFINE)` macro if feeling like doing this extra step.

Reply:
As far as I know. The esp32s3 is the only MCU jet that implements this peripheral and it implements only one instance. 
Therefore I keep it like this for now.

Reply:
(sorry for the long delay) Thank you good to note about it. This makes a lot of sense.